### DISCUSSION TITLE
Getting started with rules - very simple question

### DISCUSSION THREAD

**Discussion by BobFrankston** [arendst/Tasmota#14652] [Q&A] [answered]
I have a simple rule
```
rule1 ON button#state==0 DO webquery http://172.20.0.124/relay/0/?turn=off get ENDON ON button#state==1 DO webquery http://172.20.0.124/relay/0/?turn=on get ENDON
```

It's a wall switch from athom.tech that toggles and displays a status light. I sent to send a message when it is turned on or off. The webquery works fine standalone. So I'm missing something trivial but don't know how to trace for debug. I did issues  "rule 1" to, I think, enable the rule.

What is the obvious that I'm missing and how does one debug a rule other than trying variations?

**Comment by pkkrusty** [comment]
== compares text. I think you need =0

Also probably smart to say button1#state just to define it clearly. 

Turn on a rule is `rule1 1`

**Reply by sfromis** [reply]
It works the opposite way. Using `==` makes it a numeric comparison, while `=` is text. However, text comparison usually works for numeric, unless there are details like calculations providing values like `1.000` which is ==1 but not =1.

**Comment by sfromis** [comment]
For a Button config, you should use `Button1` instead of `Button`, the index is mandatory, not just a question of "clearly".

**Comment by BobFrankston** [comment]
Thanks, I now have
```
rule1 ON button1#state=0 DO webquery http://172.20.0.124/relay/0/?turn=off get ENDON ON button1#state=1 DO webquery http://172.20.0.124/relay/0/?turn=on get ENDON

rule1 1
```
but it's still not working. Is there a way to trace what is happening with the rule? What are best practices for debugging?

**Reply by sfromis** [reply]
A lot depends on your config. Did you configure the input as a Button? What additional config did you do?

You can check if there are any Button1 events with a test rule like `on Button1#state do var1 Button1 state is %value% endon`, and if you activate `Weblog 3` you get more details in the log.

**Reply by pkkrusty** [reply]
Also try wrapping the command in backlog, that seems to help with certain commands.

**Comment by BobFrankston** [comment, marked as answer]
Thanks.

The advice about weblog has been the key. I can now start the process of getting a deeper understanding.  As an FYI, the device is https://www.athom.tech/blank-1/us-key-switch-1-gang. When I use weblog it shows the button state is always becoming 2. But at least I'm on the path to figuring this out.

What might help most are examples of rules. There are some in the doc but the more I have the more I can understand the documentation.

**Reply by barbudor** [reply]
2 means TOGGLE, which is standard for a push button

The rules you posted seems ok.
Can you paste here the result of command `Rule0` ?
We may see something not obvious for you.

**Reply by BobFrankston** [reply]
The good news is that
```
rule1 ON power1#state=0 DO webquery http://172.20.0.124/relay/0/?turn=off get ENDON ON power1#state=1 DO webquery http://172.20.0.124/relay/0/?turn=on get ENDON
```
is working. I typed "rule1 1" to make sure in case the rule didn't come up live. Interesting is that it is showing the button state is now. I n the documentation there are some special handling I see depending on the variable.

Having something working is the major step towards figuring out more. In particular, I want to have the LED track the state of the light by sending the switch a message and then avoiding cascaded responses. 

Again, thanks for pointing me the right direction.
